pkg/controller/manage/manage_controller.go

```diff
@@ -90,107 +91,29 @@ func NewKmeshManageController(client kubernetes.Interface, security *kmeshsecuri
-				return
-			}
-			namespace, err := namespaceLister.Get(newPod.Namespace)
-			if err != nil {
-				log.Errorf("failed to get pod namespace %s: %v", newPod.Namespace, err)
-				return
-			}
-			// enable kmesh manage
-			if oldPod.Annotations[constants.KmeshRedirectionAnnotation] != "enabled" && utils.ShouldEnroll(newPod, namespace) {
-				sendCertRequest(security, newPod, kmeshsecurity.ADD)
-				if !isPodReady(newPod) {
-					log.Debugf("Pod update event: %s/%s is not ready, skipping Kmesh manage enable", newPod.GetNamespace(), newPod.GetName())
-					return
-				}
-				log.Infof("%s/%s: enable Kmesh manage", newPod.GetNamespace(), newPod.GetName())
-				nspath, _ := ns.GetPodNSpath(newPod)
-				if err := utils.HandleKmeshManage(nspath, true); err != nil {
-					log.Errorf("failed to enable Kmesh manage")
-					return
-				}
-				queue.AddRateLimited(QueueItem{podName: newPod.Name, podNs: newPod.Namespace, action: ActionAddAnnotation})
-			}
-
-			// disable kmesh manage
-			if oldPod.Annotations[constants.KmeshRedirectionAnnotation] == "enabled" && !utils.ShouldEnroll(newPod, namespace) {
-				sendCertRequest(security, oldPod, kmeshsecurity.DELETE)
-				if !isPodReady(newPod) {
-					log.Debugf("Pod update event: %s/%s is not ready, skipping Kmesh manage disable", newPod.GetNamespace(), newPod.GetName())
-					return
-				}
-				log.Infof("%s/%s: disable Kmesh manage", newPod.GetNamespace(), newPod.GetName())
-				nspath, _ := ns.GetPodNSpath(newPod)
-				if err := utils.HandleKmeshManage(nspath, false); err != nil {
-					log.Errorf("failed to disable Kmesh manage")
-					return
-				}
-				queue.AddRateLimited(QueueItem{podName: newPod.Name, podNs: newPod.Namespace, action: ActionDeleteAnnotation})
-			}
+			handlePodUpdateFunc(oldObj, newObj, namespaceLister, queue, security)
 		},
 		DeleteFunc: func(obj interface{}) {
-			pod, ok := obj.(*corev1.Pod)
-			if !ok {
-				tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-				if !ok {
-					log.Errorf("couldn't get object from tombstone %#v", obj)
-					return
-				}
-				pod, ok = tombstone.Obj.(*corev1.Pod)
-				if !ok {
-					log.Errorf("tombstone contained object that is not a Job %#v", obj)
-					return
-				}
-			}
-			if pod.Annotations[constants.KmeshRedirectionAnnotation] == "enabled" {
-				log.Infof("%s/%s: Pod managed by Kmesh is deleted", pod.GetNamespace(), pod.GetName())
-				sendCertRequest(security, pod, kmeshsecurity.DELETE)
-				// We donot need to do handleKmeshManage for delete, because we may have no change to execute a cmd in pod net ns.
-				// And we have done this in kmesh-cni
-			}
+			handlePodDeleteFunc(obj, security)
 		},
 	}); err != nil {
 		return nil, fmt.Errorf("failed to add event handler to podInformer: %v", err)
 	}
 
+	if _, err := namespaceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(Obj interface{}) {
+			handleNamespaceAddFunc(Obj, podLister, queue, security)
```

Comment:
I think we donot need to handle ns add event. As if there are pods in the ns, we must receive pod add event alongside